core/io/resource_loader.cpp

```diff
@@ -454,6 +454,30 @@ Ref<Resource> ResourceLoader::load(const String &p_path, const String &p_type_hi
 	return res;
 }
 
+// Returns a shallow reference of the loading resource.
+// Will fail if the resource is not loading.
+Ref<Resource> ResourceLoader::get_loading_resource(const String &p_path, Error *r_error) {
+	ERR_FAIL_COND_V(r_error == nullptr, nullptr);
+	*r_error = OK;
+
+	String local_path = _validate_local_path(p_path);
+	{
+		MutexLock thread_load_lock(thread_load_mutex);
+
+		Ref<Resource> existing = ResourceCache::get_ref(local_path);
+		if (existing.is_valid()) {
+			return existing;
+		}
```

Comment:
@dalexeev @RandomShaper I added a cache check here.

Reply:
In my opinion, if this new method deals with resources that are being loaded at the moment, the cache check is not relevant here. It's the caller the one responsible for checking the resource cache if finding the resource there is OK to it.

Reply:
@RandomShaper Is there a way to prevent race conditions? Like that the cache gets filled after the check of it, but before this function call?

Reply:
That said, I'm wondering if, considering the concern of this PR is a single load graph (regardless what other loads started by other threads are doing), dealing with that is necessary at all.

Reply:
@RandomShaper I'll remove the cache part, we'll reinstate it if there's issues.